Mathlib/LinearAlgebra/SesquilinearForm.lean

```diff
@@ -798,4 +798,64 @@ end CommRing
 
 end Nondegenerate
 
+namespace BilinForm
+
+lemma inner_inner_smul_sub_eq [CommRing R] [AddCommGroup M] [Module R M]
+    (B : LinearMap.BilinForm R M) (x y : M) :
+    B ((B x y) • x - (B x x) • y) ((B x y) • x - (B x x) • y) =
+      (B x x) * ((B x x) * (B y y) - (B x y) * (B y x)) := by
+  simp only [LinearMap.map_sub, LinearMap.map_smul, LinearMap.sub_apply, LinearMap.smul_apply,
+    smul_eq_mul, mul_sub, mul_assoc]
+  rw [mul_comm (B x y) (B x x), mul_left_comm (B x y) (B x x)]
+  abel
+
```

Comment:
From here everything uses the same setup so let's pull out all the `variables`:
```suggestion

variable [LinearOrderedCommRing R] [AddCommGroup M] [Module R M] (B : LinearMap.BilinForm R M)

```
Note also that `Nontrivial R` (included below, possibly not used) follows from `LinearOrderedCommRing R` so we can omit it.